### PR TITLE
Add support for System.IdentityModel.Tokens.Jwt 5.2.1

### PR DESCRIPTION
**Pull request by wowong**
1. Add support for System.IdentityModel.Tokens.Jwt 5.2.1
2. Update .csproj to the latest format.

### CODE REVIEW

File: src/Microsoft.O365.ActionableMessages.Utilities/ActionableMessageTokenValidationResult.cs [hunk 1]
```diff
@@ -1,68 +1,68 @@
...
+// THE SOFTWARE IS PROVIDED ""AS IS"", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+namespace Microsoft.O365.ActionableMessages.Utilities
```
Comment (shjin) [inline src/Microsoft.O365.ActionableMessages.Utilities/ActionableMessageTokenValidationResult.cs:29]: Why did you change the namespace? It will break the sample or existing code that uses this NuGet package.

Comment (wowong) [reply src/Microsoft.O365.ActionableMessages.Utilities/ActionableMessageTokenValidationResult.cs:29]: I figured out the namespace in the v1 package was wrong. This is a breaking change anyway due to System.IdentityModels.Tokens.Jwt 5.2.1 is not compatible with v4.0

File: src/Microsoft.O365.ActionableMessages.Utilities/Microsoft.O365.ActionableMessages.Utilities.csproj [hunk 2]
```diff
@@ -1,87 +1,23 @@
...
+    <Authors>Microsoft</Authors>
+    <Version>2.0.0</Version>
+    <Copyright>Copyright (c) Microsoft Corporation. All rights reserved.</Copyright>
+    <PackageProjectUrl>https://github.com/OfficeDev/o365-actionable-messages-utilities-for-dotnet</PackageProjectUrl>
+    <RepositoryUrl>https://github.com/OfficeDev/o365-actionable-messages-utilities-for-dotnet</RepositoryUrl>
+    <RepositoryType>GitHub</RepositoryType>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.IdentityModel.Tokens" Version="5.2.1" />
```
Comment (shjin) [inline src/Microsoft.O365.ActionableMessages.Utilities/Microsoft.O365.ActionableMessages.Utilities.csproj]: Does this work on Visual Studio 2015? If not, can you also update the README.MD of this project?

File: test/Microsoft.O365.ActionableMessages.Utilities.Tests/Microsoft.O365.ActionableMessages.Utilities.Tests.csproj [hunk 3]
```diff
@@ -1,143 +1,41 @@
...
-    <Compile Include="JsonWebToken.cs" />
-    <Compile Include="TokenValidatorTests.cs" />
-    <Compile Include="Properties\AssemblyInfo.cs" />
+    <Folder Include="Data\" />
   </ItemGroup>
+
   <ItemGroup>
-    <None Include="Data\OpenIDConfiguration.json">
+    <ProjectReference Include="..\..\src\Microsoft.O365.ActionableMessages.Utilities\Microsoft.O365.ActionableMessages.Utilities.csproj" />
+    <ProjectReference Include="..\Microsoft.O365.ActionableMessages.Utilities\Microsoft.O365.ActionableMessages.Utilities.csproj" />
```
Comment (shjin) [inline test/Microsoft.O365.ActionableMessages.Utilities.Tests/Microsoft.O365.ActionableMessages.Utilities.Tests.csproj]: One of these two references are wrong.

File: src/Microsoft.O365.ActionableMessages.Utilities/ActionableMessageTokenValidationResult.cs (hunk 1 above)
Comment (shjin) [reply src/Microsoft.O365.ActionableMessages.Utilities/ActionableMessageTokenValidationResult.cs:29]: Then please create a plan to make needed changes in below project as well:
https://github.com/OfficeDev/outlook-actionable-messages-csharp-token-validation

File: src/Microsoft.O365.ActionableMessages.Utilities/Microsoft.O365.ActionableMessages.Utilities.csproj (hunk 2 above)
Comment (wowong) [reply src/Microsoft.O365.ActionableMessages.Utilities/Microsoft.O365.ActionableMessages.Utilities.csproj]: The new .csproj only supported by VS2017. Added README.MD to both projects.

File: test/Microsoft.O365.ActionableMessages.Utilities.Tests/Microsoft.O365.ActionableMessages.Utilities.Tests.csproj (hunk 3 above)
Comment (wowong) [reply test/Microsoft.O365.ActionableMessages.Utilities.Tests/Microsoft.O365.ActionableMessages.Utilities.Tests.csproj]: Good catch!

**Review by shjin — APPROVED** [review]

File: src/Microsoft.O365.ActionableMessages.Utilities/ActionableMessageTokenValidationResult.cs (hunk 1 above)
Comment (wowong) [reply src/Microsoft.O365.ActionableMessages.Utilities/ActionableMessageTokenValidationResult.cs:29]: Will do.
